### DISCUSSION TITLE
Deserialise model from string

### DISCUSSION THREAD

**Discussion by JP-Ellis** [pydantic/pydantic#9930] [Question] [answered]
## Summary

Is it possible to have a model be deserialised from a string, as opposed to objects/dicts?

## Background

I'm going to use an example class which was only one compulsory field, which is used in another model:

```python
class Example(BaseModel):
    name: str
    version: list[int] | None = None
    attributes: list[str] | None = None

class Parent(BaseModel):
    foo: int
    bar: str
    example: Example
```

In an ideal world, I would like to support loading from both of these variants:

1. ```json
   {
     "foo": 1,
     "bar": "Lorem Ipsum",
     "example": { "name": "hello, world!" }
   }
   ```

1. ```json
   {
     "foo": 1,
     "bar": "Lorem Ipsum",
     "example": "hello, world!"
   }
   ```

Is it possible to modify the `Example` class to support both of these options?

I understand this scenario is (probably?) possible through the use of validators on the `Parent` class, but I would ideally like the `Example` class to be interchangeable with a string everywhere.

**Comment by uriyyo** [comment, marked as answer]
Hi @JP-Ellis,

You can try to use before model validator, here is example:
```py
from typing import Any

from pydantic import BaseModel, model_validator


class Example(BaseModel):
    name: str
    version: list[int] | None = None
    attributes: list[str] | None = None

    @model_validator(mode="before")
    @classmethod
    def _pre_validate(cls, value: Any) -> Any:
        if isinstance(value, str):
            return {"name": value}

        return value


class Parent(BaseModel):
    foo: int
    bar: str
    example: Example


parent = Parent.model_validate(
    {
        "foo": 1,
        "bar": "Lorem Ipsum",
        "example": "hello, world!"
    }
)
print(parent.model_dump_json(indent=4))
```
Result
```py
{
    "foo": 1,
    "bar": "Lorem Ipsum",
    "example": {
        "name": "hello, world!",
        "version": null,
        "attributes": null
    }
}
```

**Reply by JP-Ellis** [reply]
Amazing! Thank you. Solution is rather obvious in hindsight 😆
